R/checkers.R

```diff
@@ -317,3 +326,173 @@ assert_single_value <- function(x, arg = "x",
+#'
+#' @return NULL invisible
+assert_not_empty <- function(x, arg = "x",
+                             call = rlang::caller_env(),
+                             add_err_msg = "") {
+  if (nrow(x) == 0) {
+    cli::cli_abort(
+      c(
+        "{.arg {arg}} is empty",
+        add_err_msg
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+
+#' Assert that the vector of dates being passed in contains dates for each day
+#'
+#' @param dates the vector of dates to check, must be of Date type
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_daily_data <- function(dates,
+                              call = rlang::caller_env(),
+                              add_err_msg = "") {
+  sorted_dates <- dates[order(dates)]
+  # Generate a sequence of dates from the minimum to the
+  # maximum date in the dataset
+  expected_dates <- seq.Date(
+    from = min(sorted_dates),
+    to = max(sorted_dates),
+    by = "day"
+  )
+
+
+  if (!all(expected_dates %in% sorted_dates)) {
+    cli::cli_abort(
+      c(
+        "Vector of dates does not contain dates for each day",
+        add_err_msg
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+#' Assert that the vector of dates contains the duration of the specified
+#' calibration time
+#'
+#' @param date_vector the vector of dates to check, must be of Date type
+#' @param calibration_time integer indicating the number of days that
+#' the dates must span
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL invisible
+assert_sufficient_days_of_data <- function(date_vector,
+                                           calibration_time,
+                                           call = rlang::caller_env(),
+                                           add_err_msg = "") {
+  # check that you have sufficient count data for the calibration time
+  min_date <- max(date_vector,
+    na.rm = TRUE
+  ) - lubridate::days(calibration_time) + 1
+  check_sufficient_data <- min(date_vector, na.rm = TRUE) <= min_date
+  if (!check_sufficient_data) {
+    cli::cli_abort(
+      c(
+        "Insufficient data for specified calibration time"
+      ),
+      call = call,
+      class = "wwinference_specification_error"
+    )
+  }
+  invisible()
+}
+
+#' Assert that a vector of dates we're testing overlaps with the comparison
+#' dates
+#'
+#' @param test_dates the vector of dates to check, must be in IS08 convention
+#' @param comparison_dates the vector of dates to compare to, must be in ISO8
+#' convention
```

Comment:
```suggestion
#' @param comparison_dates the vector of dates to compare to, must be in ISO8601 YY-MM-DD format
```

Reply:
I'm not sure this is true? You don't call `lubridate::ymd` on the dates under the hood. Should you?

Reply:
I can add this.

Reply:
Or actually, to keep this general I don't think we need that, we just want to check that they are dates. Added an `assert_date()`